tools/bin/ecflow/processes/main/post_processing/swe_plot.py

```diff
@@ -0,0 +1,207 @@
+args = parser.parse_args()
+config_dict = read_config(args.config_file[0].name)
+
+#read in from configuration file
+direc = config_dict['VIC2NC']['OutputDirNC']
+N = config_dict['ECFLOW']['Met_Delay']
+cdf_loc = config_dict['PLOT']['cdf_SWE']
+plot_loc = config_dict['PLOT']['plot_SWE']
+
+N = int(N)
+#how many days behind metdata is from realtime
+
+date_unformat = datetime.now() - timedelta(days=N)
+
+date = date_unformat.strftime('%Y-%m-%d')
+date_ncfile = date_unformat.strftime('%Y%m%d')
+month_day = date_unformat.strftime("%B_%-d")
+
+
+
+#read VIC output that has been converted to a netcdf file 
+file = "pnw_daily_run.%s-%s.nc" %(date_ncfile, date_ncfile)
+
+ds = xr.open_dataset(os.path.join(direc,file))
+
+
+#create list of all latitudes and longitudes for full rectangle
+un_lat = ds['lat'].values
+un_lon = ds['lon'].values
+
+num_lat = len(un_lat)
+num_lon = len(un_lon)
+
+latitude = np.repeat(un_lat,num_lon)
+longitude = np.tile(un_lon, num_lat)
+
+
+#Weibull Plotting Position
+q = []
+
+for i in range(1,151):
+    q.append(i/(150.0+1))
+   
+
+min_q = min(q)/2
+max_q = max(q)+min_q
+
+
+#select out time and variable data 
+ds_day = ds.sel(time=date)
+swe_ds = ds_day['SWE']
+
+
+#create a dictionary containing the lat, lon and corresponding percentile value (if one exists)
+d = []
+
+for i in range(0,len(latitude)):
+    
+    #iterate through all latitudes and longitudes
+    ds_lat = swe_ds.sel(lat=latitude[i])
+    ds_lon = ds_lat.sel(lon=longitude[i])
+    
+    value = ds_lon.values
+
+    lat = latitude[i]
+    lon = longitude[i]
+    #read in sorted list of cdf values  
+    #if cdf cannot be read then that lat lon is saved in the dictionary without a percentile
+    #this is done to make creating the xarray dataset easier later on
+    try: 
+	cdf = pd.read_csv('%s/%s/%s_%s' %(cdf_loc, month_day, lat, lon), 
+                          index_col=None, delimiter=None, header=None)
+        x = cdf[0]
+        
+	#10mm threshold
+        if (value < 10):
+	    combine = (lat, lon)
+            d.append(combine)
+        else:    
+        
+            try:
+                #interpolation
+                f = interp1d(x,q)
+                f(value)
+                percentile = f(value)
+                combine = (lat, lon, float(percentile))
+                d.append(combine)
+            #if interpolation fails then a value is assigned based on whether it is higher or lower than the range
+            except ValueError:
+                if (value > max(x)):
+                    percentile = (max_q)
+                    combine = (lat, lon, percentile)
+                    d.append(combine)
+                else:
+                    percentile = (min_q)
+                    combine = (lat, lon, percentile)
+                    d.append(combine)
+    except IOError: 
+	percentile = value
+        combine = (lat, lon)
+        d.append(combine)
```

Comment:
I'm pretty sure lines 104-135 can be cleaned up to make this section more readable.